crates/precompiles/src/account_keychain/mod.rs

```diff
@@ -466,6 +475,14 @@ impl AccountKeychain {
             return Ok(());
         }
 
+        // Only apply spending limits if the caller is the tx origin.
+        if self.storage.spec().is_allegro_moderato() {
+            let tx_origin = self.storage.tx_origin();
+            if account != tx_origin {
+                return Ok(());
+            }
+        }
```

Comment:
Same thing here, should this return an error or are we purposely returning early?